modules/hal_nordic/nrfs/dvfs/ld_dvfs_handler.h

```diff
@@ -20,12 +20,27 @@ extern "C" {
  * @return EBUSY                  Frequency change in progress.
  * @return EAGAIN                 DVFS init in progress.
  * @return ENXIO                  Not supported frequency settings.
+ * @return EEXIST                 Requested same frequency setting as current one.
  * @return NRFS_SUCCESS           Request sent successfully.
  * @return NRFS_ERR_INVALID_STATE Service is uninitialized.
  * @return NRFS_ERR_IPC           Backend returned error during request sending.
  */
 int32_t dvfs_service_handler_change_freq_setting(enum dvfs_frequency_setting freq_setting);
 
+/**
+ * @brief Type to use as callback function in dvfs service
+ *
+ * @param params *dvfs_frequency_setting current_freq_setting
+ */
+typedef void (*dvfs_service_handler_callback)(void *params);
```

Comment:
What is actually the benefit of using a pointer here? Couldn't this value be passed directly?

Reply:
This gives more flexibility, anything can be passes here if needed.

Reply:
I don't quite understand this flexibility. The current API description specifies that this parameter points to `dvfs_frequency_setting`. How would the client know that something else was actually passed and it should be interpreted differently?

Reply:
So we have one of 3 options:
1. Change description to just pointer to parameters structure (then we can send here anything, for example in future, without changing api)
2. Pass it as copy, so we have value of dvfs_frequency_setting
3. Remove params and just use void, if it is not needed.

Which one You think is the best ?

Reply:
Fixed